packages/nestjs-auth-verify/src/services/auth-verify.service.ts

```diff
@@ -0,0 +1,204 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { AuthVerifyServiceInterface } from '../interfaces/auth-verify.service.interface';
+import { AuthVerifySettingsInterface } from '../interfaces/auth-verify-settings.interface';
+import { AuthVerifyOtpServiceInterface } from '../interfaces/auth-verify-otp.service.interface';
+import { AuthVerifyUserLookupServiceInterface } from '../interfaces/auth-verify-user-lookup.service.interface';
+import { AuthVerifyUserMutateServiceInterface } from '../interfaces/auth-verify-user-mutate.service.interface';
+import {
+  AUTH_VERIFY_MODULE_SETTINGS_TOKEN,
+  AUTH_VERIFY_MODULE_OTP_SERVICE_TOKEN,
+  AUTH_VERIFY_MODULE_USER_LOOKUP_SERVICE_TOKEN,
+  AUTH_VERIFY_MODULE_USER_MUTATE_SERVICE_TOKEN,
+  AUTH_VERIFY_MODULE_ENTITY_MANAGER_PROXY_TOKEN,
+} from '../auth-verify.constants';
+import { AuthVerifyNotificationService } from './auth-verify-notification.service';
+import {
+  ReferenceAssigneeInterface,
+  ReferenceIdInterface,
+} from '@concepta/nestjs-common';
+import { EntityManagerProxy } from '@concepta/typeorm-common';
+import { AuthVerifySendParamsInterface } from '../interfaces/auth-verify-send-params.interface';
+import { AuthVerifyConfirmParamsInterface } from '../interfaces/auth-verify-confirm-params.interface';
+import { AuthVerifyRevokeParamsInterface } from '../interfaces/auth-verify-revoke-params.interface';
+import { AuthVerifyValidateParamsInterface } from '../interfaces/auth-verify-validate-params.interface';
+
+@Injectable()
+export class AuthVerifyService implements AuthVerifyServiceInterface {
+  constructor(
+    @Inject(AUTH_VERIFY_MODULE_SETTINGS_TOKEN)
+    private readonly config: AuthVerifySettingsInterface,
+    @Inject(AUTH_VERIFY_MODULE_OTP_SERVICE_TOKEN)
+    private readonly otpService: AuthVerifyOtpServiceInterface,
+    @Inject(AUTH_VERIFY_MODULE_USER_LOOKUP_SERVICE_TOKEN)
+    private readonly userLookupService: AuthVerifyUserLookupServiceInterface,
+    @Inject(AUTH_VERIFY_MODULE_USER_MUTATE_SERVICE_TOKEN)
+    private readonly userMutateService: AuthVerifyUserMutateServiceInterface,
+    private readonly notificationService: AuthVerifyNotificationService,
```

Comment:
this constructor param should use @Inject for the service and use the interface for the type